misk-service/src/main/kotlin/misk/CoordinatedService.kt

```diff
@@ -135,11 +135,13 @@ internal class CoordinatedService(
 
   private fun startIfReady() {
     synchronized(this) {
-      if (state() != State.STARTING || service.state() != State.NEW) return
```

Comment:
This call to service.state() causes the service to be created, before doing the upstream services check below